src/app/Components/ArtworkFilter/ArtworkFilterHelpers.ts

```diff
@@ -547,3 +547,20 @@ export const getSelectedFiltersCounts = (selectedFilters: FilterArray) => {
 
   return counts
 }
+
+export const getFilterArrayFromQueryParams = (queryParams: {
+  [key: string]: string | number | boolean | string[]
+}) => {
+  return entries(queryParams).map(([key, value]) => {
+    if (value) {
+      const filterData: FilterData = {
+        // @ts-expect-error TODO: fix type error
+        displayText: FilterDisplayName[key],
+        paramName: key as FilterParamName,
+        paramValue: value,
+      }
+
+      return filterData
+    }
+  }) as FilterArray
```

Comment:
Because with `if (value)` we could have `undefined` in the array.

```suggestion
  return compact(entries(queryParams).map(([key, value]) => {
    if (value) {
      const filterData: FilterData = {
        // @ts-expect-error TODO: fix type error
        displayText: FilterDisplayName[key],
        paramName: key as FilterParamName,
        paramValue: value,
      }

      return filterData
    }
  })) as FilterArray
```

Reply:
good suggestion!